ring/pool.go

```diff
@@ -0,0 +1,61 @@
+package ring
+
+import (
+	"fmt"
+
+	"github.com/tuneinsight/lattigo/v6/utils/structs"
+)
+
+// Pool represents a pool of polys that can be used (concurrently) to instantiate temporary polynomials.
+type Pool struct {
```

Comment:
This type name is a bit vague (a pool is a generic concept). `PolyPool` or `BufferPool` or `MemPool` ? Other suggestions welcome.

Reply:
I guess `PolyPool` makes sense at this level but not at higher level where there are other types of pooled values like ciphertexts.

We might also want to reflect that on some field names (but less a priority as most are private anyway).